x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -598,32 +598,40 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
   );
 
   const editRule = useMemo(() => {
-    if (!hasActionsPrivileges) {
-      return (
-        <EuiToolTip position="top" content={ruleI18n.EDIT_RULE_SETTINGS_TOOLTIP}>
-          <LinkButton
-            onClick={goToEditRule}
-            iconType="controlsHorizontal"
-            isDisabled={true}
-            href={formatUrl(getEditRuleUrl(ruleId ?? ''))}
-          >
-            {ruleI18n.EDIT_RULE_SETTINGS}
-          </LinkButton>
-        </EuiToolTip>
-      );
-    }
-    return (
+    const renderEditRuleSettingButtonLink = (disabled: boolean) => (
       <LinkButton
-        data-test-subj="editRuleSettingsLink"
+        data-test-subj={disabled ? undefined : 'editRuleSettingsLink'}
         onClick={goToEditRule}
         iconType="controlsHorizontal"
-        isDisabled={!isExistingRule || !userHasPermissions(canUserCRUD)}
+        isDisabled={disabled}
         href={formatUrl(getEditRuleUrl(ruleId ?? ''))}
       >
         {ruleI18n.EDIT_RULE_SETTINGS}
       </LinkButton>
     );
-  }, [isExistingRule, canUserCRUD, formatUrl, goToEditRule, hasActionsPrivileges, ruleId]);
+
+    if (!hasActionsPrivileges || (isMlRule(rule?.type) && !hasMlPermissions)) {
+      return (
+        <EuiToolTip
+          position="top"
+          content={getToolTipContent(rule, hasMlPermissions, hasActionsPrivileges)}
+        >
+          {renderEditRuleSettingButtonLink(true)}
+        </EuiToolTip>
+      );
+    }
+
+    return renderEditRuleSettingButtonLink(!isExistingRule || !userHasPermissions(canUserCRUD));
```

Comment:
Out of the scope of this PR, but it would also be great to show a tooltip message when `!userHasPermissions(canUserCRUD)` is true.

Reply:
that part is fixed as well